packages/mui-material/src/Select/Select.d.ts

```diff
@@ -148,9 +148,45 @@ export interface SelectProps<Value = unknown>
    * The variant to use.
    * @default 'outlined'
    */
-  variant?: 'standard' | 'outlined' | 'filled';
+  variant?: 'filled' | 'standard' | 'outlined';
 }
 
+export interface FilledSelectProps extends Omit<FilledInputProps, 'value' | 'onChange'> {
+  /**
+   * The variant to use.
+   * @default 'outlined'
+   */
+  variant: 'filled';
+}
+
+export interface StandardSelectProps extends Omit<InputProps, 'value' | 'onChange'> {
+  /**
+   * The variant to use.
+   * @default 'outlined'
+   */
+  variant: 'standard';
+}
+
+export interface OutlinedSelectProps extends Omit<OutlinedInputProps, 'value' | 'onChange'> {
+  /**
+   * The variant to use.
+   * @default 'outlined'
+   */
+  variant: 'outlined';
+}
+
+export type SelectVariants = 'outlined' | 'standard' | 'filled';
+
+export type SelectProps<
+  Value = unknown,
+  Variant extends SelectVariants = SelectVariants,
+> = BaseSelectProps<Value> &
+  (Variant extends 'filled'
+    ? FilledSelectProps
+    : Variant extends 'standard'
+    ? StandardSelectProps
+    : OutlinedSelectProps);
```

Comment:
This is a little rough for people using `SelectProps` directly because the `variant` prop is required now when it wasn't before. I guess using `BaseSelectProps` is a suitable replacement for most use cases?

Reply:
Thanks! Using `BaseSelectProps` instead of `SelectProps` fixed the build in our component library after updating mui to 5.15.11

Reply:
Fixed it in https://github.com/mui/material-ui/pull/41359